### PR TITLE
Remove MachineDefines.h

### PR DESCRIPTION
**Pull request by zdroid**
This is a move towards more idiomatic file structure. Currently the linking of source files is a mess, and removing one that's not needed will reduce the mess.

### CODE REVIEW

File: src/Constants.h [hunk 1]
```diff
@@ -63,6 +63,9 @@ constexpr double EARTH_DIAMETER = EARTH_CIRCUMFERENCE / PI;
  */
 constexpr double EARTHRADIUS = EARTH_DIAMETER / 2;
 
+#define MAX_NUM_THREADS 96
+#define CACHE_LINE_SIZE 64
+
```
Comment (matt-gretton-dann) [inline src/Constants.h:68]: As you're moving these - make them into `const int` please.

Comment (zdroid) [reply src/Constants.h]: Done

Comment (zdroid) [reply src/Constants.h]: ping @matt-gretton-dann

**Comment by weshinsley** [comment]
@matt-gretton-dann - also we should update .vcxproj file when this is merged.

**Comment by zdroid** [comment]
@weshinsley that change already is in this PR

**Comment by weshinsley** [comment]
Requested changes were addressed - simple changed - resolved and merged.

**Comment by zdroid** [comment]
@weshinsley Great work! You may want to use squash or rebase instead of merge, as it doesn't make a merge commit.

**Comment by weshinsley** [comment]
Ooh right - I'm not sure how to do that from your branch -  I've been strictly following the command-line instructions when github can't do the conflict resolution on the page...

**Comment by zdroid** [comment]
@weshinsley I deleted my repository along with this and the other branches, so it probably made the job harder for you. My last 3 PRs should be fine regarding that.

**Pull request closed without merging**